src/Spago/Version.hs

```diff
@@ -70,37 +70,32 @@ getNextVersion spec version@SemVer{..} =
 
 
 -- | Make a tag for the new version.
-tagNewVersion :: Spago m => DryRun -> SemVer -> SemVer -> m ()
-tagNewVersion dryRun oldVersion newVersion = do
+tagNewVersion :: Spago m => SemVer -> SemVer -> m ()
+tagNewVersion oldVersion newVersion = do
 
   let oldVersionTag = unparseVersion oldVersion
       newVersionTag = unparseVersion newVersion
 
-  case dryRun of
-    DryRun -> do
-      echo $ "Skipped creating new Git tag (" <> newVersionTag <> ") because this is a dry run."
-    NoDryRun -> do
-      Git.commitAndTag newVersionTag $ oldVersionTag <> " → " <> newVersionTag
-      echo $ "Git tag created for new version: " <> newVersionTag
+  Git.commitAndTag newVersionTag $ oldVersionTag <> " → " <> newVersionTag
+  echo $ "Git tag created for new version: " <> newVersionTag
 
 
 -- | Bump and tag a new version in preparation for release.
-bumpVersion :: Spago m => Maybe Int -> DryRun -> VersionBump -> m ()
-bumpVersion limitJobs dryRun spec = do
-  DryRun.showHelp dryRun
-
-  Git.requireCleanWorkingTree
```

Comment:
I'm not sure about removing `Git.requireCleanWorkingTree`, since (at least without other changes to compensate), it results in misleading output.

If I run (on this branch) `spago bump-version patch`, and I have some uncommitted changes, then Spago says `spago: A new bower.json has been generated. Please commit this and run bump-version again.` ... but actually no new `bower.json` was generated :stuck_out_tongue: